tripy/tripy/function_registry.py

```diff
@@ -371,6 +371,21 @@ def impl(func):
             # based on argument types.
             if isinstance(func, property):
                 self[key] = func
+            # For classes, we apply the wrapper to all methods.
+            elif inspect.isclass(func):
+                for member in dir(func):
+                    # Ignore properties and functions not defined in the class (we will use the presence of a docstring as a proxy for that).
+                    # It does not suffice to check just that the method is inherited because some decorators like @dataclass add methods
+                    # that are not documented or annotated and do not use inheritance to do so.
+                    # We need to check hasattr first because dir can list some properties even if they're not defined (e.g.,  __abstractmethods__).
```

Comment:
Would `inspect.getmembers(func, predicate=lambda obj: inspect.isfunction(obj) and obj.__doc__)` work better here?

Reply:
Oh that's cool. Yeah, it works and is more compact.